src/google-maps/map-heatmap-layer/map-heatmap-layer.ts

```diff
@@ -0,0 +1,175 @@
+   */
+  @Input()
+  set options(options: Partial<google.maps.visualization.HeatmapLayerOptions>) {
+    this._options = options;
+  }
+  private _options: Partial<google.maps.visualization.HeatmapLayerOptions>;
+
+  /**
+   * The underlying google.maps.visualization.HeatmapLayer object.
+   *
+   * See: https://developers.google.com/maps/documentation/javascript/reference/visualization
+   */
+  heatmap?: google.maps.visualization.HeatmapLayer;
+
+  constructor(
+    private readonly _googleMap: GoogleMap,
+    private _ngZone: NgZone) {}
+
+  ngOnInit() {
+    if (this._googleMap._isBrowser) {
+      if (!window.google?.maps?.visualization && (typeof ngDevMode === 'undefined' || ngDevMode)) {
+        throw Error(
+            'Namespace `google.maps.visualization` not found, cannot construct heatmap. ' +
+            'Please install the Google Maps JavaScript API with the "visualization" library: ' +
+            'https://developers.google.com/maps/documentation/javascript/visualization');
+      }
+
+      // Create the object outside the zone so its events don't trigger change detection.
+      // We'll bring it back in inside the `MapEventManager` only for the events that the
+      // user has subscribed to.
+      this._ngZone.runOutsideAngular(() => {
+        this.heatmap = new google.maps.visualization.HeatmapLayer(this._combineOptions());
+      });
+      this._assertInitialized();
+      this.heatmap.setMap(this._googleMap.googleMap!);
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    const {_data, heatmap} = this;
+
+    if (heatmap) {
+      if (changes['options']) {
+        heatmap.setOptions(this._combineOptions());
+      }
+
+      if (changes['data'] && _data !== undefined) {
+        // TODO(crisbeto): the API allows for `MVCArray<LatLng|WeightedLocation>` or
+        // `Array<LatLng|WeightedLocation>`, however the typings are incorrect and they only allow
+        // `MVCArray<LatLng|WeightedLocation>` or `Array<LatLng>` or `Array<WeightedLocation>` so
+        // we have to cast to `any` here. We should remove the cast once the typings have been fixed
+        // See: https://developers.google.com/maps/documentation/javascript/reference/visualization
+        heatmap.setData(this._normalizeData(_data) as any);
+      }
+    }
+  }
+
+  ngOnDestroy() {
+    if (this.heatmap) {
+      this.heatmap.setMap(null);
+    }
+  }
+
+  /**
+   * Gets the data that is currently shown on the heatmap.
+   * See: developers.google.com/maps/documentation/javascript/reference/visualization#HeatmapLayer
+   */
+  getData(): HeatmapData {
+    this._assertInitialized();
+    return this.heatmap.getData();
+  }
+
+  /** Creates a combined options object using the passed-in options and the individual inputs. */
+  private _combineOptions(): google.maps.visualization.HeatmapLayerOptions {
+    return {
+      ...(this._options || {}),
+      data: this._normalizeData(this._data || []),
```

Comment:
I may be reading this wrong, but it seems like a user could not set data in the "options" object this way. For example, if a user uses the "options" input that includes "data", then it would be overwritten by an empty array. Is this correct? Is it intended?

Reply:
You're right and it's unintended. I've submitted a fix to have it fall back to the data from the options first.